src/ModelContextProtocol.Core/Authentication/ClientOAuthProvider.cs

```diff
@@ -193,39 +193,56 @@ private async Task PerformOAuthAuthorizationAsync(
         CancellationToken cancellationToken)
     {
         // Get available authorization servers from the 401 response
-        var protectedResourceMetadata = await ExtractProtectedResourceMetadata(response, _serverUrl, cancellationToken).ConfigureAwait(false);
-        var availableAuthorizationServers = protectedResourceMetadata.AuthorizationServers;
-
-        if (availableAuthorizationServers.Count == 0)
+        ProtectedResourceMetadata? protectedResourceMetadata = null;
+        try
         {
-            ThrowFailedToHandleUnauthorizedResponse("No authorization servers found in authentication challenge");
+            protectedResourceMetadata = await ExtractProtectedResourceMetadata(response, _serverUrl, cancellationToken).ConfigureAwait(false);
         }
-
-        // Select authorization server using configured strategy
-        var selectedAuthServer = _authServerSelector(availableAuthorizationServers);
-
-        if (selectedAuthServer is null)
+        catch
         {
-            ThrowFailedToHandleUnauthorizedResponse($"Authorization server selection returned null. Available servers: {string.Join(", ", availableAuthorizationServers)}");
+            protectedResourceMetadata = null;
```

Comment:
We shouldn't blindly catch all exceptions. For example, this swallows the `throw McpException($"Failed to fetch resource metadata")` this PR adds to ExtractProtectedResourceMetadata even though this should be enough to cause McpClient.CreateAsync to throw due to an authentication failure.